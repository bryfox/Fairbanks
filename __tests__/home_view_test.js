import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import HomeView from '../src/presentation/home_view'

const EmptyForecast = {
  details: []
}
const EmptyState = {
  refreshing: false,
  forecast: {
      Today: EmptyForecast,
      Extended: EmptyForecast,
      Recreational: EmptyForecast
    }
}

describe('HomeView', () => {
  const noop = () => {}

  it('renders links to other screens', () => {
    const tree = renderer.create(
      <HomeView
          isRefreshing={false}
          didPressExtended={noop}
          didPressRecreational={noop}
          didPressWeb={noop}
          didRequestRefresh={noop}
          forecasts={EmptyState.forecast}
        />
    )
    expect(tree).toMatchSnapshot()
  })

})
