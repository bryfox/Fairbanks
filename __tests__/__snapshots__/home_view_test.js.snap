// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView renders links to other screens 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
  >
    <RCTRefreshControl />
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "margin": 20,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#666",
              },
              Object {
                "flex": 1,
                "fontSize": 24,
                "marginBottom": 20,
                "marginTop": 20,
              },
            ]
          }
        >
          Forecast Unavailable
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "textAlign": "center",
            }
          }
        >
          Today’s forecast isn’t available yet.
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "textAlign": "center",
            }
          }
        >
          Check back later!
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Extended Forecast"
      accessibilityTraits={
        Array [
          "button",
          "disabled",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Extended
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Recreational Forecast"
      accessibilityTraits={
        Array [
          "button",
          "disabled",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Recreational
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Web Version"
      accessibilityTraits={
        Array [
          "button",
          "disabled",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
            Object {
              "padding": 8,
            },
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#dfdfdf",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "Entypo",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
